router/core/graphql_prehandler.go

```diff
@@ -597,6 +602,36 @@ func (h *PreHandler) handleOperation(req *http.Request, buf *bytes.Buffer, httpO
 
 	httpOperation.traceTimings.EndPlanning()
 
+	/**
+	* Validate that the planned query doesn't exceed the maximum query depth configured
+	 */
+
+	if h.maxQueryDepth > 0 {
+		_, maxQueryDepthSpan := h.tracer.Start(req.Context(), "Operation - Checking Query Depth",
+			trace.WithSpanKind(trace.SpanKindInternal),
+			trace.WithAttributes(otel.WgEngineRequestTracingEnabled.Bool(httpOperation.traceOptions.Enable)),
+			trace.WithAttributes(attributes...),
+		)
+
+		operation := opContext.preparedPlan.operationDocument
+		definition := opContext.preparedPlan.schemaDocument
+		report := operationreport.Report{}
+		globalComplexityResult, _ := operation_complexity.CalculateOperationComplexity(operation, definition, &report)
+
+		if globalComplexityResult.Depth > h.maxQueryDepth {
+			err := &httpGraphqlError{
```

Comment:
You also need to the set the finErr so the root span is marked as failed.

Reply:
Any error returned from this function is set as the finErr

Reply:
Great, I couldn't see this in the diff.